Prime 9ja Online.js

```diff
@@ -0,0 +1,313 @@
+*/
+
+function detectWeb(doc, _url) {
+	let jsonLdNodes = doc.querySelectorAll('script[type="application/ld+json"]');
+	for (let node of jsonLdNodes) {
+		try {
+			let data = JSON.parse(node.textContent);
+			let type = data['@type'];
+			if (typeof type === 'string' && type.endsWith('NewsArticle')) {
+				return 'newspaperArticle';
+			}
+			if (Array.isArray(type) && type.some(t => typeof t === 'string' && t.endsWith('NewsArticle'))) {
+				return 'newspaperArticle';
+			}
+		}
+		catch (e) {
+			// ignore JSON parsing errors
+		}
+	}
+
+	if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	let rows = doc.querySelectorAll('a.entry-title[href*="/202"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) === 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url = doc.location.href) {
+	let item = new Zotero.Item('newspaperArticle');
+	let jsonLdNodes = doc.querySelectorAll('script[type="application/ld+json"]');
+	let data = null;
+
+	for (let node of jsonLdNodes) {
+		try {
+			let parsed = JSON.parse(node.textContent);
+			let type = parsed['@type'];
+			if (
+				type && typeof type === 'string' && type.endsWith('NewsArticle')
+				|| Array.isArray(type) && type.some(t => typeof t === 'string' && t.endsWith('NewsArticle'))
+			) {
+				data = parsed;
+				break;
+			}
+		}
+		catch (e) {}
+	}
+
+	if (data) {
+		item.title = data.headline || ZU.xpathText(doc, '//h1[contains(@class, "entry-title")]');
+		item.ISSN = '3092-8907';
+		item.abstractNote = data.description || '';
+		item.date = data.datePublished || '';
+		item.language = data.inLanguage || 'en';
+		item.url = data.url || url;
+		item.publicationTitle = (data.publisher && data.publisher.name) || 'Prime 9ja Online';
+		item.place = 'Nigeria';
+
+		if (data.author) {
+			if (Array.isArray(data.author)) {
+				for (let author of data.author) {
+					if (author.name) {
+						item.creators.push(ZU.cleanAuthor(author.name, 'author'));
+					}
+				}
+			}
+			else if (data.author.name) {
+				item.creators.push(ZU.cleanAuthor(data.author.name, 'author'));
+			}
+		}
+		else {
+			let authorText = ZU.xpathText(doc, '//span[@itemprop="name"]');
+			if (authorText) {
+				item.creators.push(ZU.cleanAuthor(authorText, 'author'));
+			}
+		}
+
+		if (data.thumbnailUrl) {
+			item.attachments.push({
+				title: 'Image',
+				mimeType: 'image/jpeg',
+				url: data.thumbnailUrl
+			});
+		}
```

Comment:
We're saving a snapshot, so this can be removed. We only save images when the item is of type `artwork` or it's e.g. an archived newspaper scan that's only available as an image.